truthy-falsy.js

```diff
@@ -0,0 +1,219 @@
+// 1
+function setAlarm(isEmployed, isOnVacation){
+  return isEmployed && !isOnVacation;
+}
+console.log(setAlarm(true, false));
+
+// 2
+function closeCompare(firstNumber, secondNumber, numbersMargin = 0){
+  const numbersDiff = Math.abs(firstNumber - secondNumber);
+  if (numbersDiff <= numbersMargin) {
+    return 0;
+  }
+  if (firstNumber < secondNumber) {
+    return -1;
+  }
+  return 1;
+}
+
+// 3
+function or(firstBoolean, secondBoolean) {
+  if (firstBoolean === true) {
+    return true;
+  }
+  if (secondBoolean === true) {
+    return true;
+  }
+  return false;
+}
+console.log(or(true, false));
+
+function xor(firstVariable, secondVariable) {
+  if (firstVariable && !secondVariable) {
+    return true;
+  }
+  if (secondVariable && !firstVariable) {
+    return true;
+  }
+  return false;
+}
+console.log(xor(true, true));
+
+// 4
+function isTruthy(booleanVariable, isBooleanTruth, isBooleanFalse) {
+  if (booleanVariable === true) {
+    return isBooleanTruth();
+  }
+  return isBooleanFalse();
+}
+
+// 5
+function rentalCarCost(days) {
+  const baseCost = days * 40;
+  if (days >= 7) {
+    return baseCost - 50;
+  }
+  if (days >= 3) {
+    return baseCost - 20;
+  }
+  return days * 40;
+}
+
+// 6
+function getRealFloor(floorNumber) {
+  if (floorNumber <= 0) {
+    return floorNumber;
+  }
+  if (floorNumber >= 13) {
+    return floorNumber - 2;
+  }
+  return floorNumber - 1;
+}
+
+// 7
+function willRockWin(opponent) {
+  return opponent === 'lizard' || opponent === 'scissors';
+}
+function willPaperWin(opponent) {
+  return opponent === 'rock' || oponent === 'spock';
+}
+function willScissorsWin(opponent) {
+  return opponent === 'paper' || opponent === 'lizard';
+}
+function willLizardWin(opponent) {
+  return opponent === 'spock' || opponent === 'paper';
+}
+function willSpockWin(opponent) {
+  return opponent === 'scissors' || 'rock';
+}
+
+function rockPaperScissorsLizardSpock(playerOne, playerTwo) {
+  if (playerOne === playerTwo) {
+    return 'Draw!';
+  }
+  if (playerOne === 'rock' && willRockWin(playerTwo)) {
+    return 'Player 1 Won!';
+  }
+  if (playerOne === 'paper' && willPaperWin(playerTwo)) {
+    return 'Player 1 Won!';
+  }
+  if (playerOne === 'scissors' && willScissorsWin(playerTwo)) {
+    return 'Player 1 Won!';
+  }
+  if (playerOne === 'lizard' && willLizardWin(playerTwo)) {
+    return 'Player 1 Won!';
+  }
+  if (playerOne === 'spock' && willSpockWin(playerTwo)) {
+    return 'Player 1 Won!';
+  }
+  return 'Player 2 Won!';
+}
+
+// 8
+function isValidPassword(password) {
+  const specialCharacters = /[`!@#$%^&*()_\-+=\[\]{};':"\\|,.<>\/?~ ]/;
+  const passwordLengthCheck = password.trim().length >= 8;
+  const passwordCapitalCharactersCheck = /[A-Z]/.test(password) >= 1;
+  const passwordSpecialCharactersCheck = specialCharacters.test(password) >= 1;
```

Comment:
Those names don't suggest that those are booleans.
For example change `specialCharacters` to `containsSpecialCharacters`